### PR TITLE
Reoders fields to match data set. Normalizes all names to match columns.

### PR DESCRIPTION
**Pull request by cheradenine**
This PR does the following:
* Reorders all the fields to be identical to what is in the source dataset.
* Changes all the names to be consistent with the abbreviations
* * We had some that were modified to be more readable but for data debugging it will be easier if they are the same
* Adds comment to the data key document
* Changes some of the string fields to limit number of chars (for single letter codes)
* Convert values that are in the dataset as 'Y' or 'N' to boolean - will need to handle this conversion on the data import

### CODE REVIEW

File: backend/db/migrate/20220817011608_create_property_assessments.rb [hunk 1]
```diff
@@ -1,69 +1,77 @@
...
-
-      t.string :owner
-      t.string :mail_addressee
-      t.string :mail_address
-      t.string :mail_city
-      t.string :mail_state, limit: 2
-      t.string :mail_zipcode # these can be 5 or zip+4 format
-      t.float :res_floor
+      t.string  :lu, limit: 2
+      t.string  :luc_desc
```
Comment (jonnycodesalot) [inline backend/db/migrate/20220817011608_create_property_assessments.rb:22]:
```suggestion
      t.string  :lu_desc
```

Typo I think - it's lu_desc in the spreadsheet

**Review by jonnycodesalot — APPROVED** [review]
Thanks for doing this!  Definitely much easier to digest with them all matching.  I don't like that there are a few fields where the meaning isn't obvious to me from the name, but I don't feel strongly about which is the right way to go there. This definitely is a big improvement regardless :-)

**Review by AlisonPAnderson — APPROVED** [review]
Thank you John!  This does help a lot!

**Comment by jonnycodesalot** [comment]
I actually just found I was unable to perform db:migrate with this branch, where I was previously.  Not sure why yet, but i rescinded the approvals :-/

**Review by jonnycodesalot — APPROVED** [review]

**Pull request merged**